src/transaction/boot_sr.c

```diff
@@ -2229,6 +2229,13 @@ boot_restart_server (THREAD_ENTRY * thread_p, bool print_restart, const char *db
 
   init_server_type (db_name);
 
+  if (get_server_type () == SERVER_TYPE_PAGE && !HA_DISABLED ())
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_INCOMPATIBLE_SERVER_TYPE_HA_CONFIG, 0);
+      error_code = ER_FAILED;
+      goto error;
+    }
```

Comment:
Nitpicking: let error_code be as specific as possible.
```suggestion
  if (get_server_type () == SERVER_TYPE_PAGE && !HA_DISABLED ())
    {
      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_INCOMPATIBLE_SERVER_TYPE_HA_CONFIG, 0);
      error_code = ER_INCOMPATIBLE_SERVER_TYPE_HA_CONFIG;
      goto error;
    }
```